src/index.ts

```diff
@@ -196,12 +196,17 @@ class S3Storage extends StorageBase {
         const stream = output.Body as Readable
         stream.pipe(res)
       } catch (err) {
-        res.status(404)
-        next(err)
+        if (err.name === 'NoSuchKey') {
+          res.status(404).send('Image not found');
+        } else {
+          res.status(500);
+          next(err);
+        }
```

Comment:
I haven't checked it against Ghost source, but are we supposed to finalize the response in this handler by using `send`? If so, should we also finalize it in `500` case?

Reply:
Good question! Generally, I would expect Ghost to handle it, but I have just done some tests which would indicate the opposite.

Once I finalize the response with the `.send()`, there is no timeout.

So, the simplest solution to the timeout error would indeed be removal of the `next(err)` and implementation of `send()`.

In my eyes, the 500 error does, however, not accurately reflect the issue at hand – the missing key.

After thinking about this once again, the more elegant (and extendable) solution would probably be this:

```
} catch (err) {
    switch (err.name) {
        case 'NoSuchKey':
            res.status(404).send('Specified key does not exist');
            break;
        default:
            res.status(500).send('Internal server error');
    }
}
```

Let me know what you think. Happy to adapt :)

Reply:
I just checked [Ghost's own local file storage adapter](https://github.com/TryGhost/Ghost/blob/78d2a5e3c094ae4e1ec463ac1e64a1ebe949d83b/ghost/core/core/server/adapters/storage/LocalStorageBase.js#L144-L150), it seems we should call `next()` with the error type defined in `@tryghost/errors`.

I think we should use that approach instead of finalizing the response.